Week5/Solution.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Alyssa Lingad
+ * 
+ * MICP: Week 4
+ */
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import java.util.*;
+
+
+
+public class Solution
+{
+	public static boolean isPalindrome(Node head) throws Exception{
+		if (head == null){
+			throw new Exception();
+		}
+		Node copy = copyLL(head);
+		int nodeCount = 0;
+		
+		Node prev = null;
+		Node curr = copy;
+		Node next = null;
+		while (curr != null){
+			next = curr.next;
+			curr.next = prev;
+			prev = curr;
+			curr = next;
+			nodeCount++;
+		}
+		copy = prev;
+		
+		int stoppingPoint = nodeCount / 2;
+
+		while (stoppingPoint > 0){
+			if (copy.value != head.value){
+				return false;
+			}
+			copy = copy.next;
+			head = head.next;
+			stoppingPoint--;
+		}
+		return true;
+	}
```

Comment:
Instead of cloning the entire input and reversing it, and using only half of the orginal and clone for comparison, you can try to just reverse the second half or first half of the original link list.